packages/react-aria-components/docs/Virtualizer.mdx

```diff
@@ -0,0 +1,430 @@
+            <Row style={{width: 'inherit', height: 'inherit'}}>
+              <Cell><MyCheckbox slot="selection" /></Cell>
+              <Cell>{item.foo}</Cell>
+              <Cell>{item.bar}</Cell>
+              <Cell>{item.baz}</Cell>
+            </Row>
+          )}
+        </TableBody>
+      </Table>
+    </Virtualizer>
+  );
+}
+```
+
+```css hidden
+.react-aria-Table {
+  width: 400;
+  height: 300;
+  overflow: auto;
+  scroll-padding-top: 38px;
+}
+
+.react-aria-TableHeader {
+  background: var(--spectrum-gray-100);
+  width: 100%;
+  height: 100%;
+  border-radius: 8px;
+  font-weight: bold;
+}
+
+.react-aria-Cell,
+.react-aria-Column {
+  display: flex;
+  align-items: center;
+  height: 100%;
+  box-sizing: border-box;
+}
+```
+
+## Advanced: Custom layouts
+
+Custom Virtualizer layouts can be created by extending the `Layout` abstract base class. At a minimum, the `getVisibleLayoutInfos`, `getLayoutInfo`, and `getContentSize` methods must be implemented. You can override the other methods to customize their default behavior.
+
+<TypeContext.Provider value={docs.links}>
+  <InterfaceType {...docs.exports.Layout} hideProperties />
+</TypeContext.Provider>
+
+### LayoutInfo
+
+Layouts produce `LayoutInfo` objects describing the position, size, and other properties of each item in a collection. Virtualizer requests this information when needed, and uses it to create DOM nodes to display.
+
+<TypeContext.Provider value={docs.links}>
+  <InterfaceType {...docs.exports.LayoutInfo} hideMethods />
+</TypeContext.Provider>
+
+### Example
+
+This example implements a horizontally scrolling layout with fixed size items.
+
+```tsx example
+import {Layout, LayoutInfo, Rect, Size, Key} from 'react-aria-components';
+
+class HorizontalLayout extends Layout {
+  // Determine which items are visible within the given rectangle.
+  getVisibleLayoutInfos(rect: Rect): LayoutInfo[] {
+    let keys = Array.from(this.virtualizer!.collection.getKeys());
+    let startIndex = Math.max(0, Math.floor(rect.x / 100));
+    let endIndex = Math.min(keys.length - 1, Math.ceil(rect.maxX / 100));
+    let layoutInfos = [];
+    for (let i = startIndex; i <= endIndex; i++) {
+      layoutInfos.push(this.getLayoutInfo(keys[i]));
+    }
+    return layoutInfos;
+  }
+
+  // Provide a LayoutInfo for a specific item.
+  getLayoutInfo(key: Key): LayoutInfo | null {
+    let node = this.virtualizer!.collection.getItem(key);
+    if (!node) {
+      return null;
+    }
+
+    let rect = new Rect(node.index * 100, 0, 100, 100);
+    return new LayoutInfo(node.type, node.key, rect);
+  }
+
+  // Provide the total size of all items.
+  getContentSize(): Size {
+    let numItems = this.virtualizer!.collection.size;
+    return new Size(numItems * 100, 100);
+  }
+}
+
+function Example() {
+  let layout = React.useMemo(() => new HorizontalLayout(), []);
+
+  let items = [];
+  for (let i = 0; i < 200; i++) {
+    items.push({id: i, name: `Item ${i}`});
+  }
+
+  return (
```

Comment:
![Screenshot 2025-02-05 at 8 09 52 am](https://github.com/user-attachments/assets/cc3f76d9-170d-41ac-b0a7-37d8b9b627b3)
Last example needs to have less height on the items so it doesn't vertically scroll

Reply:
lol, stupid scrollbars 😂